ui/src/services/QueueService.ts

```diff
@@ -1,70 +1,228 @@
+  }
 }
 
-function trimAttributeValue(attributeName: string, attributeValue: string) {
-    switch (attributeName) {
-        case "CreatedTimestamp":
-        case "LastModifiedTimestamp":
-            return new Date(parseInt(attributeValue) * 1000).toISOString();
-        case "RedrivePolicy":
-            const redriveAttributeValue: QueueRedrivePolicyAttribute = JSON.parse(attributeValue)
-            const deadLetterTargetArn = "DeadLetterTargetArn: " + redriveAttributeValue.deadLetterTargetArn
-            const maxReceiveCount = "MaxReceiveCount: " + redriveAttributeValue.maxReceiveCount
-            return deadLetterTargetArn + ", " + maxReceiveCount
-        default:
-            return attributeValue;
+function parseReceiveMessageResponse(xmlData: string): QueueMessage[] {
+  try {
+    const messages: QueueMessage[] = [];
+
+    const messageRegex = /<Message>([\s\S]*?)<\/Message>/g;
+    const messageIdRegex = /<MessageId>([\s\S]*?)<\/MessageId>/;
+    const receiptHandleRegex = /<ReceiptHandle>([\s\S]*?)<\/ReceiptHandle>/;
+    const bodyRegex = /<Body>([\s\S]*?)<\/Body>/;
+    const sentTimestampRegex =
+      /<Name>SentTimestamp<\/Name>\s*<Value>([\s\S]*?)<\/Value>/;
+
+    let match;
+    while ((match = messageRegex.exec(xmlData)) !== null) {
+      const messageXml = match[1];
+
+      const messageIdMatch = messageIdRegex.exec(messageXml);
+      const receiptHandleMatch = receiptHandleRegex.exec(messageXml);
+      const bodyMatch = bodyRegex.exec(messageXml);
+      const sentTimestampMatch = sentTimestampRegex.exec(messageXml);
+
+      if (messageIdMatch && bodyMatch) {
+        messages.push({
+          messageId: messageIdMatch[1],
+          receiptHandle: receiptHandleMatch ? receiptHandleMatch[1] : undefined,
+          body: bodyMatch[1],
+          sentTimestamp: sentTimestampMatch
+            ? sentTimestampMatch[1]
+            : new Date().toISOString(),
+          attributes: {},
+          messageAttributes: {},
+        });
+      }
+    }
+
+    return messages;
+  } catch (error) {
+    console.error("Error parsing XML response:", error);
+    return [];
+  }
+}
 
+async function deleteMessage(
+  queueName: string,
+  messageId: string,
+  receiptHandle: string
+): Promise<void> {
+  try {
+    const params = new URLSearchParams();
+    params.append("Action", "DeleteMessage");
+    params.append("ReceiptHandle", receiptHandle);
+
+    const response = await sqsInstance.post(`queue/${queueName}`, params, {
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    });
+
+    if (response.status !== 200) {
+      throw new Error(`Failed to delete message: ${response.statusText}`);
     }
+  } catch (error) {
+    console.error(
+      `Error deleting message ${messageId} from ${queueName}:`,
+      error
+    );
+    throw error;
+  }
 }
 
-export default {getQueueListWithCorrelatedMessages, getQueueAttributes}
\ No newline at end of file
+export default {
```

Comment:
I think it's better to export these functions separately and import the whole service like this:

`import * as QueueService from 'path/to/queue/service'`;

or just to import the methods we want to use